tests/src/test/java/com/maxio/advancedbilling/utils/TestTeardown.java

```diff
@@ -30,19 +31,29 @@ public class TestTeardown {
     private final AdvancedBillingClient advancedBillingClient = TestClient.createClient();
 
     public void deleteCustomer(Customer customer) throws IOException, ApiException {
+        deleteCustomer(customer, true);
+    }
+
+    public void deleteCustomer(Customer customer, boolean failWhenNoExists) throws IOException, ApiException {
```

Comment:
Get rid of this flag and check response code - ignore exception if this is 404, rethrow otherwise.